math/src/unsigned_integer/element.rs

```diff
@@ -478,17 +478,26 @@ impl<const NUM_LIMBS: usize> ByteConversion for UnsignedInteger<NUM_LIMBS> {
     fn from_bytes_be(bytes: &[u8]) -> Result<Self, ByteConversionError> {
         let mut limbs = Vec::with_capacity(NUM_LIMBS);
 
-        bytes[0..NUM_LIMBS * 8]
-            .chunks_exact(8)
-            .try_for_each(|chunk| {
-                let limb = u64::from_be_bytes(
-                    chunk
-                        .try_into()
-                        .map_err(|_| ByteConversionError::FromBEBytesError)?,
-                );
-                limbs.push(limb);
-                Ok::<_, ByteConversionError>(())
-            })?;
+        // We cut off extra bytes, this is useful when you use this function to generate the element from randomness
+        // In the future with the right algorithm this shouldn't be needed
+
+        let needed_bytes_res = bytes.get(0..NUM_LIMBS * 8);
+
+        if needed_bytes_res.is_none() {
+            return Err(ByteConversionError::FromBEBytesError);
+        }
+
+        let needed_bytes = needed_bytes_res.unwrap();
```

Comment:
nit(non-blocking): you can make this more brief as follows:
```suggestion
        let needed_bytes = bytes.get(0..NUM_LIMBS * 8)
            .ok_or(ByteConversionError::FromBEBytesError)?;
```

Reply:
This way you could also keep the chain of composition if you want.